_posts/2021-08-23-gsoc-2021-project.md

```diff
@@ -0,0 +1,116 @@
+
+## Contents
+
+- [Visualizing the Performance Characteristics of Computations](#visualizing-the-performance-characteristics)
+- [Graphical Representation of Task Graphs](#graphical-representation)
+- [HTML Representation of Task Graphs](#html-representation)
+
+## Visualizing the Performance Characteristics of Computations
+
+## Graphical Representation of Task Graphs
+
+I worked on enhancing the graphviz output of the task graphs. The Graphviz library boasts a great set of attributes which can be modifified to create a more visually appealing output. I worked on making them more illustrative, engaging, and informative.
+
+### Fixing calling `.visualize()` with `filename=None` [#7740](https://github.com/dask/dask/pull/7740)
+
+I fixed a minor bug that was caused when users tried to call `dask.visualize()` with `filename=None`. I fixed it by adding an extra condition before it reaches the error line to make sure the format is not `None` by making it equal to the default `png`.
+
+```python
+import dask
+import dask.array as da
+
+array = da.arange(10)
+dask.visualize(array, filename=None)
+```
+
+<img src="/images/gsoc21/7740.png" alt="#7740 Demo" height=414 width=736>
+
+### Add node size scaling to the Graphviz output for the high level graphs [#7869](https://github.com/dask/dask/pull/7869)
+
+I tweaked the node sizes in the High-level graph's task graphs to align with the density of each layer. Layers with more tasks would appear larger than the rest.
+
+```python
+import dask.array as da
+
+array = da.random.random((10000, 10000), chunks=(200, 200))
+result = array + array.T - array.mean(axis=0)
+
+result.dask.visualize()
+```
+
+<img src="/images/gsoc21/7869.png" alt="#7869 Demo" height=414 width=736>
+
+### Change graphviz font family to sans [#7931](https://github.com/dask/dask/pull/7931)
+
+I modified the font of the Graphviz graphs to `Helvetica` (a sans-serif fontface). This is cleaner and more readable than the default `Times-Roman` (a serif fontface).
+
+```python
+import dask.array as da
+
+array = da.ones((10, 10), chunks=(5, 5))
+array = array + 100
+array = array * 100
+
+array.dask.visualize()
+```
+
+<img src="/images/gsoc21/7931.png" alt="#7931 Demo" height=414 width=736>
+
+### Add tooltips to graphviz [#7973](https://github.com/dask/dask/pull/7973)
+
+I added tooltips to the High-level graph's task graphs. The tooltips show the `layer_type`, `number of tasks`, and the information stored in the `collection_annotations` dictionary for each layer.
+
+```python
+import dask.array as da
+
+x = da.ones((10, 10), chunks=(5, 5))
+x = x + 100
+x = x * 100
+
+x.dask.visualize()
+```
+
+<img src="/images/gsoc21/7973.png" alt="#7973 Demo" height=414 width=736>
+
+### Add colors to represent high level layer types [#7974](https://github.com/dask/dask/pull/7974)
+
+<img src="/images/gsoc21/7974.png" alt="#7974 Demo" height=414 width=736>
+
+## HTML Representation of Task Graphs
+
```

Comment:
Extra references we can link to:
* Blogpost introducing the high level graph HTML repr in July this year - https://blog.dask.org/2021/07/07/high-level-graphs#visualization
* Matt's blogpost about HTML reprs from 2019 - https://matthewrocklin.com/blog/2019/07/04/html-repr

Reply:
Also for the distributed side of things, we'll want to acknowledge Jacob Tomlinson's work (eg: https://github.com/dask/distributed/pull/4857 and https://github.com/dask/distributed/pull/4853)